### PR TITLE
Add doctest example for broadcast!

### PR DESCRIPTION
**Pull request by kshyatt**
Just wanted to show the different `dest` behaviors.

### CODE REVIEW

**Comment by kshyatt** [comment]
Looks like we had a build timeout on OSX. Any thoughts on this?

**Comment by KristofferC** [comment]
My thought is that building GCC from scratch on CI is an unwise thing to do :)

**Comment by kshyatt** [comment]
@KristofferC you're paying the snarky comment review price 😛. Does the added example seem ok?

File: base/broadcast.jl [hunk 1]
```diff
@@ -713,6 +713,32 @@ Like [`broadcast`](@ref), but store the result of
 Note that `dest` is only used to store the result, and does not supply
 arguments to `f` unless it is also listed in the `As`,
 as in `broadcast!(f, A, A, B)` to perform `A[:] = broadcast(f, A, B)`.
+
+# Examples
+```jldoctest
+julia> A = [1.0; 0.0];
+
+julia> B = [0.0; 0.0];
```
Comment (KristofferC) [inline base/broadcast.jl]: Put `A` and `B` on the same line?

I noticed more and more that is becomes kinda annoying when the examples take too much vertical space so I try to condense as much as possible.

**Review by KristofferC — APPROVED** [review]

**Pull request merged**